File: seriesData/series.py
from seriesData.season import Season


class Series:
    def __init__(self, name: str) -> None:
        self.name = name
        self.seasons = list()
        self.image = ""
        self.desc = ""
        self.link = ""


    def addSeason(self, episodes, name=None):
        self.seasons.append(Season(episodes, name))
        

    def removeSeason(self, season):
        self.seasons.remove(season)
        

File: src/data/series.ts
import { Season } from "./season";

export class Series {
    id:number;
    name:string;
    seasons: Season[];
    image:string;
    desc:string;
    link:string

    constructor(id:number, name:string) {
        this.id = id;
        this.name = name;
        this.seasons = [];
        this.image = "";
        this.desc = "";
        this.link = "";
    }

    addSeason(episodes:number, name:string|undefined=undefined) {
        this.seasons.push(new Season(name, episodes));
    }

    removeSeason(season:Season) {
        let index = this.seasons.indexOf(season);
        if(index == -1) return
        this.seasons.splice(index,1);
    }
}